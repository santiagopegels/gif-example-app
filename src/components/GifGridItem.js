import React from 'react'

export const GifGridItem = ({ id, title, url }) => {
    return (
        <div>
            <img src={url} />
            <p>{title}</p>
        </div>
    )
}
