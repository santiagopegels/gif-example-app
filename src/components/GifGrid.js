import React from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGif } from '../hooks/useFetchGif';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGif(category);

    return (
        <>
            <h3>{category}</h3>

            {loading && <p>Loading...</p>}

            <div className="card-grid">
                {
                    images.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
