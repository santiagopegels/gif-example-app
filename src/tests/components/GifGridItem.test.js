import React from 'react'
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Test <GifGridItem />', () => {

    const title = "Test"
    const url = "https://localhost/test.html"
    const wrapper = shallow(<GifGridItem title={title} url={url} />)


    test('Should show the component', () => {

        expect(wrapper).toMatchSnapshot();
    })


    test('Should show an paragraph element with title prop', () => {

        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })

    test('Should be the same url and alt props in image element', () => {

        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)

    })

    test('Should have animate__fadeIn class', () => {

        const div = wrapper.find('div')
        const className = div.prop('className')
        expect(className.includes('animate__fadeIn')).toBe(true)

    })
})
