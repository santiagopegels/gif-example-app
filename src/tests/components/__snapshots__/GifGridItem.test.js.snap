// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem /> Should show the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Test"
    src="https://localhost/test.html"
  />
  <p>
    Test
  </p>
</div>
`;
